### ISSUE TITLE
Try training to detect binary captcha and got 0.00% IOU and recall

### ISSUE THREAD

**Issue by happy15** [AlexeyAB/darknet#181]
(Seems here is more popular than the darknet google group.)

First I am quite new to yolo and darknet and machine learning.

I tried to detect binary captcha like the one below, which consists of upper and lower case letters and numbers (in all 62 classes):

![1](https://user-images.githubusercontent.com/983570/29869953-c3b45eb4-8db7-11e7-878f-ae7b465e538e.jpg)

I prepared 1000 labeled images (yes according to what @AlexeyAB suggested this is a lot less than 62*2000=124000) and used a dummy network cfg (see the end). The 1st batch resulted in 0.02 avg and after about 3000 batach it decreased to 0.016130 avg and seems stops decreasing. 

As suggested, I stoped and tested recall on the **training data**, 0.00% IOU and recall are what I got.

My question is, is it merely because the training dataset is too small? or is this problem a good use case for darknet / cnn since it lacks of texture and color which I think benefited the imagenet detection?

The cfg:

```
[net]
batch=128
#batch=1
subdivisions=1
height=128
width=128
channels=3
momentum=0.9
decay=0.0005
max_crop=128

learning_rate=0.1
policy=poly
power=4
max_batches=160000

[convolutional]
batch_normalize=1
filters=16
size=7
stride=1
pad=3
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=62
size=3
stride=1
pad=1
activation=linear

[avgpool]

[softmax]
groups=1

[cost]
type=sse
```

**Comment by AlexeyAB** [comment]
@happy15 Hi,

1. What command did you use to train this network?
    May be should be used `./darknet captcha train your.cfg initial.weights`

2. What command did you use to validation of captcha recognition?
    May be should be used `./darknet captcha valid your.cfg trained.weights`

3. What labels format of images did you use for training?

**Comment by happy15** [comment]
Thanks for response. I used detector train/recall. And the label format is relative to image width height and x,y are coresponds to the center

**Comment by AlexeyAB** [comment]
>  I used **detector** train/recall

But your cfg, from your 1st question hasn't region-layer, so it can't be used for **Detector**: https://github.com/AlexeyAB/darknet/blob/master/cfg/yolo-voc.2.0.cfg#L227

Use yolo-voc.2.0.cfg.

**Comment by happy15** [comment]
ok, I see, I'll try to add one

Alexey <notifications@github.com>于2017年8月31日周四 下午10:32写道：

**Comment by happy15** [comment]
Btw, what are all these options in region layer means? for example anchor.
Any readme I can read?

Sheng LUO <sheng.peisi.luo@gmail.com>于2017年9月1日周五 下午1:11写道：

**Comment by happy15** [comment]
Since I solved this by constructing my own resnet style cnn, this issue could be closed now.
